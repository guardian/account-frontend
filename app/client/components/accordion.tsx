import React, { ReactElement, ReactNode } from "react";
import palette from "../colours";
import { sans } from "../styles/fonts";

export interface AccordionPanelProps {
  title: string;
}

export interface AccordionProps {
  children: Array<ReactElement<AccordionPanelProps>>;
}

export interface AccordionState {
  selectedIndex: number;
}

const svgUpCaret = (
  <svg viewBox="0 0 1024 574" width="100%" height="100%">
    <path d="M1015 564q-10 10-23 10t-23-10L512 82 55 564q-10 10-23 10T9 564q-9-10-9-24t9-24L489 10q10-10 23-10t23 10l480 506q9 10 9 24t-9 24z" />
  </svg>
);

const svgDownCaret = (
  <svg viewBox="0 0 1024 574" width="100%" height="100%">
    <path d="M1015 10q-10-10-23-10t-23 10L512 492 55 10Q45 0 32 0T9 10Q0 20 0 34t9 24l480 506q10 10 23 10t23-10l480-506q9-10 9-24t-9-24z" />
  </svg>
);

export class Accordion extends React.Component<AccordionProps, AccordionState> {
  constructor(props: AccordionProps) {
    super(props);
    this.state = {
      selectedIndex: 0
    };
  }

  public render(): ReactNode {
    return (
      <div css={{ flexGrow: 0, flexShrink: 0 }}>
        {this.props.children.map(
          (child: ReactElement<AccordionPanelProps>, index: number) => (
            <React.Fragment key={child.props.title}>
              <button
                onClick={() =>
                  this.setState({
                    selectedIndex:
                      index === this.state.selectedIndex ? -1 : index
                  })
                }
                css={{
                  display: "block",
                  fontFamily: sans,
                  fontSize: "inherit",
                  backgroundColor:
                    palette.neutral[
                      index === this.state.selectedIndex ? "4" : "5"
                    ],
                  color: palette.neutral["1"],
                  cursor: "pointer",
                  width: "100%",
                  padding: "10px",
                  margin: "2px",
                  textAlign: "left",
                  border: "none",
                  outline: "none",
                  transition: "0.4s",
                  ":hover": {
                    backgroundColor: palette.neutral["4"]
                  }
                }}
              >
                <div
                  css={{
                    display: "flex",
                    justifyContent: "space-between",
                    alignItems: "center"
                  }}
                >
                  <span css={{ flexShrink: 0 }}>{child.props.title}</span>
                  <div
                    css={{
                      height: "13px",
                      width: "13px",
                      color: palette.neutral["1"],
                      marginLeft: "5px"
                    }}
                  >
                    {index === this.state.selectedIndex
                      ? svgDownCaret
                      : svgUpCaret}
                  </div>
                </div>
              </button>
              <div
                css={{
                  padding:
                    index === this.state.selectedIndex ? "5px 5px" : "0 5px",
                  marginLeft: "5px",
                  backgroundColor: palette.white,
                  overflow: "hidden",
                  maxHeight: index === this.state.selectedIndex ? undefined : 0
                }}
              >
                {child}
              </div>
            </React.Fragment>
          )
        )}
      </div>
    );
  }
}
