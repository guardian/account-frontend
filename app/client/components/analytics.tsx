import { Location } from "@reach/router";
import React, { ReactNode } from "react";

declare global {
  interface Window {
    ga?: any;
    gaData?: any;
    dataLayer?: any;
  }
}

export interface Event {
  eventCategory: string;
  eventAction: string;
  eventLabel?: string;
  eventValue?: number;
}

export const trackEvent = ({
  eventCategory,
  eventAction,
  eventLabel,
  eventValue
}: Event) => {
  if (window.ga) {
    window.ga(
      "send",
      "event",
      eventCategory,
      eventAction,
      eventLabel,
      eventValue
    );
  }
};

export const applyAnyOptimiseExperiments = () => {
  if (typeof window !== "undefined" && window.ga && window.dataLayer) {
    window.dataLayer.push({ event: "optimize.activate" });
  }
};

export class AnalyticsTracker extends React.PureComponent<{}> {
  constructor(props: {}) {
    super(props);
    if (typeof window !== "undefined" && window.ga) {
      if (window.dataLayer === undefined) {
        window.dataLayer = [];
      }

      window.ga("create", "UA-51507017-5", "auto");
      window.ga("require", "GTM-M985W29");
      window.ga("set", "transport", "beacon");

      new MutationObserver(applyAnyOptimiseExperiments).observe(document.body, {
        attributes: false,
        characterData: false,
        childList: true,
        subtree: true,
        attributeOldValue: false,
        characterDataOldValue: false
      });
    }
  }

  public render(): ReactNode {
    return (
      <Location>
        {({ location }) => {
          if (location && typeof window !== "undefined" && window.ga) {
            window.ga("send", "pageview", {
              location: location.href,
              page: location.pathname + location.search
            });
            applyAnyOptimiseExperiments();
          }
          return null; // null is a valid React node type, but void is not.
        }}
      </Location>
    );
  }
}
