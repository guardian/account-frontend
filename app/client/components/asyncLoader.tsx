import React from "react";
import Spinner from "./spinner";

export interface AsyncLoaderProps<T> {
    fetch: () => Promise<T>;
    render: (data: T) => React.ReactNode;
}

export enum LoadingState {
    loading,
    loaded,
    error
}

export interface AsyncLoaderState<T> {
    data?: T
    loadingState: LoadingState
};

export default class AsyncLoader<T> extends React.Component<
    AsyncLoaderProps<T>,
    AsyncLoaderState<T>
    > {

    state: AsyncLoaderState<T> = {loadingState: LoadingState.loading}

    public componentDidMount(): void {
        this.props.fetch().then(data => {
            this.setState({data, loadingState: LoadingState.loaded});
        }).catch(_ => this.setState({loadingState: LoadingState.error}))
    }


    public render() {

        switch (this.state.loadingState){
            case LoadingState.loading : return <Spinner />;
            case LoadingState.loaded : return this.state.data ? this.props.render(this.state.data) : <h1>Super Boom</h1>;
            case LoadingState.error : return <h1>Boom</h1>;
        }

    }
}
