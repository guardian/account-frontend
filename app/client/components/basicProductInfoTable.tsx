import React from "react";
import { DateHelper } from "../../shared/dates";
import { ProductDetail } from "../../shared/productResponse";
import { GroupedProductType } from "../../shared/productTypes";
import { ProductDescriptionListTable } from "./productDescriptionListTable";

interface BasicProductInfoTableProps {
  groupedProductType: GroupedProductType;
  productDetail: ProductDetail;
}
export const BasicProductInfoTable = (props: BasicProductInfoTableProps) => {
  return (
    <ProductDescriptionListTable
      content={[
        ...(props.groupedProductType.shouldRevealSubscriptionId
          ? [
              {
                title: "Subscription ID",
                value: props.productDetail.subscription.subscriptionId
              }
            ]
          : []),
        ...(props.groupedProductType.tierLabel
          ? [
              {
                title: props.groupedProductType.tierLabel,
                value: props.productDetail.tier
              }
            ]
          : []),
        ...(props.groupedProductType.shouldShowJoinDateNotStartDate
          ? [
              {
                title: "Join date",
                value: DateHelper(props.productDetail.joinDate).dateStr()
              }
            ]
          : [
              {
                title: "Start date",
                value: props.productDetail.subscription.start
                  ? DateHelper(props.productDetail.subscription.start).dateStr()
                  : "-"
              }
            ])
      ]}
    />
  );
};
