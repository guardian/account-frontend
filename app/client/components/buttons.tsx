import { css } from "@emotion/core";
import { Link } from "@reach/router";
import Color from "color";
import React from "react";
import palette from "../colours";
import { sans } from "../styles/fonts";
import { ArrowIcon } from "./svgs/arrowIcon";
import { TickIcon } from "./svgs/tickIcon";

export interface ButtonProps {
  text: string;
  onClick?: () => void;
  height?: string;
  fontWeight?: "bold";
  left?: true;
  right?: true;
  disabled?: boolean;
  colour?: string;
  textColour?: string;
  primary?: true;
  hollow?: true;
  hide?: boolean;
  forceCircle?: true;
  hoverColour?: string;
  leftTick?: true;
}

export interface LinkButtonProps extends ButtonProps {
  to: string;
  state?: any;
}

const applyIconStyleIfApplicable = (
  hover: boolean,
  left?: true,
  right?: true,
  leftTick?: true
) => {
  if (left) {
    return hover ? styles.leftHover : styles.left;
  } else if (right) {
    return hover ? styles.rightHover : styles.right;
  } else if (leftTick) {
    return {
      padding: "4px 21px 3px 16px"
    };
  }
  return {
    padding: "1px 15px 0 15px",
    svg: {
      display: "none"
    }
  };
};

const calcBackgroundColour = (
  disabled?: boolean,
  colour?: string,
  primary?: true,
  hollow?: true
) => {
  if (disabled) {
    return palette.neutral["4"];
  } else if (primary) {
    return palette.yellow.medium;
  } else if (hollow) {
    return palette.white;
  }
  return colour;
};

const calcTextColour = (
  disabled?: boolean,
  textColour?: string,
  primary?: true,
  hollow?: true
) => {
  if (disabled) {
    return palette.white;
  } else if (primary || hollow) {
    return palette.neutral["1"];
  }
  return textColour;
};

const defaultColour = palette.neutral["2"];
const buttonCss = ({
  disabled,
  height,
  fontWeight,
  colour = defaultColour,
  textColour = palette.white,
  left,
  right,
  primary,
  hollow,
  hide,
  forceCircle,
  hoverColour,
  leftTick
}: ButtonProps) => {
  const backgroundColour = calcBackgroundColour(
    disabled,
    colour,
    primary,
    hollow
  );
  return css({
    fontSize: "16px",
    fontFamily: sans,
    borderRadius: "1000px",
    alignItems: "center",
    position: "relative",
    ":active": {
      outline: "none"
    },
    minHeight: height || "36px",
    fontWeight,
    display: hide ? "none" : "inline-flex",
    background: backgroundColour,
    color: calcTextColour(disabled, textColour, primary, hollow),
    border: hollow ? "1px solid" : "none",
    ...applyIconStyleIfApplicable(false, left, right, leftTick),
    ...(forceCircle
      ? {
          padding: "1px 18px 0 18px"
        }
      : {}),
    ":hover": disabled
      ? undefined
      : {
          background:
            hoverColour ||
            Color(backgroundColour)
              .darken(backgroundColour === defaultColour ? 0.3 : 0.1)
              .string(),
          ...applyIconStyleIfApplicable(true, left, right, leftTick)
        },
    cursor: disabled ? "not-allowed" : "pointer",
    maxWidth: "calc(100vw - 40px)"
    // [maxWidth.mobile]: {
    //   maxWidth: maxWidthIfWrapping || "280px"
    // },
    // [maxWidth.mobileLarge]: {
    //   maxWidth: maxWidthIfWrapping || "320px",
    //   whiteSpace: "normal",
    //   lineHeight: "16px"
    // }
  });
};

const styles = {
  leftHover: {
    svg: { transform: "translate(-3px, -50%) rotate(180deg)" }
  },
  left: {
    padding: "1px 18px 0 40px",
    svg: {
      fill: "currentColor",
      height: "34px",
      position: "absolute",
      left: "0px",
      top: "50%",
      transform: "translate(0, -50%) rotate(180deg)",
      transition: "transform .3s, background .3s",
      width: "36px"
    }
  },
  rightHover: {
    svg: { transform: "translate(3px, -50%)" }
  },
  right: {
    padding: "1px 40px 0 18px",
    svg: {
      fill: "currentColor",
      height: "34px",
      position: "absolute",
      right: "0",
      top: "50%",
      transform: "translate(0, -50%)",
      transition: "transform .3s, background .3s",
      width: "36px"
    }
  }
};

export const LinkButton = (props: LinkButtonProps) => (
  <Link
    to={props.disabled ? "" : props.to}
    onClick={props.onClick}
    css={buttonCss(props)}
    state={props.state}
  >
    {props.text}
    <ArrowIcon />
  </Link>
);

export const Button = (props: ButtonProps) => (
  <button
    onClick={props.onClick}
    css={buttonCss(props)}
    disabled={props.disabled}
    onMouseUp={(event: React.MouseEvent<HTMLButtonElement>) =>
      (event.target as HTMLButtonElement).blur()
    }
  >
    {props.leftTick && <TickIcon />}
    {props.text}
    {(props.left || props.right) && <ArrowIcon />}
  </button>
);
