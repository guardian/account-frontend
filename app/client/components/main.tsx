import React from "react";
import palette from "../colours";
import { serif } from "../styles/fonts";
import { Footer } from "./footer/footer";
import Header from "./header";

export interface WithOptionalServerPathWithQueryParams {
  serverPathWithQueryParams?: string;
}

interface MainProps extends WithOptionalServerPathWithQueryParams {
  children: any;
}

export const Main = ({ children }: MainProps) => {
  const tmpIsSignedIn =
    typeof window !== "undefined" && !!window.guardian?.identityDetails?.userId;
  return (
    <div
      css={{
        display: "flex",
        flexDirection: "column",
        height: "100vh",
        alignItems: "stretch",
        width: "100%",
        color: palette.neutral["2"],
      }}
    >
      <Header isSignedOut={!tmpIsSignedIn} />
      <div
        css={{
          flexGrow: 1,
          flexShrink: 0,
          display: "flex",
          flexDirection: "column",
        }}
      >
        <main
          css={{
            fontFamily: serif,
            flexGrow: 1,
            flexShrink: 0,
          }}
        >
          {children}
        </main>
      </div>
      <Footer />
    </div>
  );
};
