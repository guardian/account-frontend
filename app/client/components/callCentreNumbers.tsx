import { css } from "@emotion/core";
import React from "react";
import { Accordion } from "./accordion";

const contactUsStyles = {
  margin: "0 0 10px",
  paddingRight: "5px"
};

const callCenterStyles = css({
  marginBottom: "10px",
  display: "flex",
  flexWrap: "wrap",
  textAlign: "left",
  fontWeight: "normal"
});

const ukPhoneCommonPart = "330 333 6790";
export const ukPhoneNumber = `+44 (0) ${ukPhoneCommonPart}`;
export const ukPhoneNumberWithoutPrefix = `0${ukPhoneCommonPart}`;
export const ukPhoneNumberWithoutPrefix = "0330 333 6790";

export const ukOpeningTimes =
  "8am - 8pm on weekdays, 8am-6pm at weekends (GMT/BST)";

export interface CallCentreNumbersProps {
  prefixText?: string;
}

export const CallCentreNumbers = (props: CallCentreNumbersProps) => (
  <div css={callCenterStyles}>
    {props.prefixText && <p css={contactUsStyles}>{props.prefixText}</p>}
    <Accordion>
      <div title="United Kingdom, Europe and rest of world">
        <div>
          <b>{ukPhoneNumber}</b>
        </div>
        <div>{ukOpeningTimes}</div>
      </div>
      <div title="Australia, New Zealand, and Asia Pacific">
        <div>
          <b>1800 773 766</b> (within Australia)
        </div>
        <div>
          <b>+61 2​ 8​076 8599</b> (outside Australia)
        </div>
        <div>9am - 5pm Monday - Friday (AEDT)</div>
      </div>
      <div title="Canada and USA">
        <div>
          <b>1-844-632-2010</b> (toll free USA)
        </div>
        <div>
          <b>+1 917-900-4663</b> (outside USA)
        </div>
        <div>9am - 5pm on weekdays (EST/EDT)</div>
      </div>
    </Accordion>
  </div>
);
