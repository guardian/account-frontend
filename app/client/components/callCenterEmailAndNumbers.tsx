import { css } from "@emotion/core";
import { palette } from "@guardian/src-foundations";
import { textSans } from "@guardian/src-foundations/typography";
import React, { useState } from "react";
import { CallCentreNumbersProps } from "./callCentreNumbers";

const contactUsStyles = {
  margin: "0 0 10px",
  paddingRight: "5px"
};

const callCenterStyles = css({
  marginBottom: "10px",
  display: "flex",
  flexWrap: "wrap",
  textAlign: "left",
  fontWeight: "normal"
});

export type PhoneRegionKey = "US" | "AUS" | "UK & ROW";

interface PhoneRegion {
  key: PhoneRegionKey;
  title: string;
  openingHours: string;
  phoneNumbers: Array<{ phoneNumber: string; suffix?: string }>;
}

const EMAIL_ADDRESS: string = "customer.help@theguardian.com";

const PHONE_DATA: PhoneRegion[] = [
  {
    key: "UK & ROW",
    title: "United Kingdom, Europe and rest of world",
    openingHours: "8am - 8pm on weekdays, 8am - 6pm at weekends (GMP/BST)",
    phoneNumbers: [
      {
        phoneNumber: "+44 (0) 330 333 6790"
      }
    ]
  },
  {
    key: "AUS",
    title: "Australia, New Zealand, and Asia Pacific",
    openingHours: "9am - 5pm Monday - Friday (AEDT)",
    phoneNumbers: [
      {
        phoneNumber: "1800 773 766",
        suffix: "(within Australia)"
      },
      {
        phoneNumber: "+61 28076 8599",
        suffix: "(outside Australia)"
      }
    ]
  },
  {
    key: "US",
    title: "Canada and USA",
    openingHours: "9am - 5pm on weekdays (EST/EDT)",
    phoneNumbers: [
      {
        phoneNumber: "1-844-632-2010",
        suffix: "(toll free USA)"
      },
      {
        phoneNumber: "+1 917-900-4663",
        suffix: "(outside USA)"
      }
    ]
  }
];

interface CallCentreEmailAndNumbersProps extends CallCentreNumbersProps {
  hideEmailAddress?: boolean;
  phoneRegionFilterKeys?: PhoneRegionKey[];
}

export const CallCentreEmailAndNumbers = (
  props: CallCentreEmailAndNumbersProps
) => {
  const [indexOfOpenSection, setIndexOfOpenSection] = useState<number>(0);

  const filteredPhoneData = PHONE_DATA.filter(
    phoneRegion =>
      !props.phoneRegionFilterKeys ||
      props.phoneRegionFilterKeys.includes(phoneRegion.key)
  );

  const sectionTitleCss = (isOpen: boolean, isNotFirstOption: boolean) => `
    ${textSans.medium()};
    margin: 0;
    padding: 12px 17px 12px 12px;
    position: relative;
    cursor: pointer;
    :after {
      content: "";
      display: block;
      width: 7px;
      height: 7px;
      border-top: 2px solid ${palette.neutral["7"]};
      border-right: 2px solid ${palette.neutral["7"]};
      position: absolute;
      top: 50%;
      transform: translateY(-50%) ${
        isOpen ? "rotate(-45deg)" : "rotate(135deg)"
      };
      transition: transform 0.4s;
      right: 17px;
    }
    ${isNotFirstOption &&
      `
      :before {
        content: "";
        display: block;
        position: absolute;
        top: 0;
        left: 12px;
        width: calc(100% - 24px);
        height: 1px;
        background-color: ${palette.neutral["86"]}
      }
    `}
  `;
  const innerSectionCss = (isOpen: boolean) => `
    display: ${isOpen ? "block" : "none"};
    background-color: ${palette.neutral["97"]};
    padding: 12px;
  `;

  const innerSectionPCss = `
    ${textSans.medium()};
  `;

  const innerSectionBlockSpanCss = `
      display: block;
      margin-bottom: 4px;
      font-weight: bold;
  `;

  const innerSectionTitleCss = `
    ${textSans.medium()};
    margin: 6px 0 4px;
  `;

  const handleSectionClick = (sectionNum: number) => () => {
    setIndexOfOpenSection(indexOfOpenSection === sectionNum ? -1 : sectionNum);
  };
  return (
    <div css={callCenterStyles}>
      {props.prefixText && <p css={contactUsStyles}>{props.prefixText}</p>}
      <div
        css={css`
          width: 100%;
          border: 1px solid ${palette.neutral["86"]};
        `}
      >
        {filteredPhoneData.map((phoneRegion, index) => {
          const isOpen = index === indexOfOpenSection;
          const isNotFirstOption = index > 0;
          return (
            <div css={css`subsectionCss`} key={phoneRegion.key}>
              <h2
                css={css`
                  ${sectionTitleCss(isOpen, isNotFirstOption)}
                `}
                onClick={handleSectionClick(index)}
              >
                {phoneRegion.title}
              </h2>
              <div
                css={css`
                  ${innerSectionCss(isOpen)}
                `}
              >
                {!props.hideEmailAddress && (
                  <>
                    <h4
                      css={css`
                        ${innerSectionTitleCss}
                      `}
                    >
                      Email:
                    </h4>
                    <span
                      css={css`
                        ${textSans.medium({ fontWeight: "bold" })};
                      `}
                    >
                      {EMAIL_ADDRESS}
                    </span>
                  </>
                )}
                <h4
                  css={css`
                    ${innerSectionTitleCss}
                  `}
                >
                  Phone:
                </h4>
                <p
                  css={css`
                    ${innerSectionPCss}
                  `}
                >
                  {phoneRegion.phoneNumbers.map(({ phoneNumber, suffix }) => (
                    <span
                      key={phoneNumber}
                      css={css`
                        ${innerSectionBlockSpanCss}
                      `}
                    >
                      {phoneNumber}
                      {suffix && (
                        <span
                          css={css`
                            font-weight: normal;
                          `}
                        >
                          {" "}
                          {suffix}
                        </span>
                      )}
                    </span>
                  ))}
                  {phoneRegion.openingHours}
                </p>
              </div>
            </div>
          );
        })}
      </div>
    </div>
  );
};
