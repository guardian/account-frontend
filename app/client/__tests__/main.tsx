import "dom-testing-library/extend-expect";
import React from "react";

import serializer from "jest-emotion";

expect.addSnapshotSerializer(serializer);

import renderer from "react-test-renderer";

import { Main } from "../components/main";

test("Main renders something", () => {
  const rendered = renderer.create(
    <Main>
      <p>hi</p>
    </Main>
  );

  expect(rendered.toJSON()).toMatchSnapshot();
});
