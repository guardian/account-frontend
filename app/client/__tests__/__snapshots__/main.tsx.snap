// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders something 1`] = `
.emotion-0 {
  width: 56px;
}

<div
  css={
    Object {
      "alignItems": "stretch",
      "color": "#333333",
      "display": "flex",
      "flexDirection": "column",
      "minHeight": "100vh",
      "width": "100%",
    }
  }
>
  <header
    css={
      Object {
        "backgroundColor": "#e9eff1",
        "borderBottomColor": "rgba(0, 0, 0, 0.14)",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "0.8px",
        "color": "#121212",
        "height": "2.8125rem",
      }
    }
  >
    <div
      css={
        Object {
          "fontSize": "0.875rem",
          "lineHeight": "2.8125rem",
          "margin": "auto",
          "maxWidth": "71.25rem",
          "paddingLeft": "1.25rem",
          "paddingRight": "1.25rem",
        }
      }
    >
      <span />
      <a
        css={
          Object {
            "float": "right",
            "padding": "2px",
          }
        }
        href="https://www.theguardian.com"
      >
        <svg
          className="emotion-0"
          viewBox="0 0 56 56"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M28 0a28 28 0 1 0 28 28A28 28 0 0 0 28 0"
            fill="#121212"
          />
          <path
            d="M33 6.92c3.63.58 8.24 3.06 9.89 4.83v8h-1L33 7.82zm-3.34.5h-.09c-6.35 0-9.8 8.8-9.8 20.66 0 11.87 3.45 20.66 9.8 20.66h.09v.91c-9.56.65-22.42-6.63-22.09-21.58C7.23 13.14 20.09 5.86 29.66 6.51zm16.16 22.53l-3 1.3v13.44A24.26 24.26 0 0 1 33 49.52V31l-3.3-1.09V29h16.12z"
            fill="#fff"
          />
        </svg>
      </a>
    </div>
  </header>
  <div
    css={
      Object {
        "flex": "1",
      }
    }
  >
    <main
      css={
        Object {
          "fontFamily": "\\"Guardian Text Egyptian Web\\",Georgia,serif",
          "margin": "auto",
          "maxWidth": "48.75rem",
          "padding": "30px",
        }
      }
    >
      <p>
        hi
      </p>
    </main>
  </div>
  <footer>
    <div
      css={
        Object {
          "backgroundColor": "#e9eff1",
          "color": "#121212",
          "height": "42px",
          "text-align": "right",
        }
      }
    >
      <div
        css={
          Object {
            "lineHeight": "42px",
            "margin": "auto",
            "maxWidth": "71.25rem",
            "paddingRight": "2rem",
          }
        }
      >
        <a
          css={
            Object {
              "color": "#121212",
            }
          }
          href="#top"
        >
          back to top
        </a>
      </div>
    </div>
    <div>
      <div
        css={
          Object {
            "backgroundColor": "#333",
            "color": "#bdbdbd",
            "fontSize": "0.75rem",
            "padding": "0.25rem",
          }
        }
      >
        <div
          css={
            Object {
              "margin": "auto",
              "maxWidth": "71.25rem",
            }
          }
        >
          © 2018 Guardian News and Media Limited or its affiliated companies. All rights reserved.
        </div>
      </div>
    </div>
  </footer>
</div>
`;
